src/audio/module_adapter/module_adapter.c

```diff
@@ -1319,6 +1388,12 @@ int module_adapter_bind(struct comp_dev *dev, void *data)
 	/* set the source dev pointer */
 	mod_source_info->sources[source_index] = source_dev;
 
+	/* Disable 1 source 1 sink simple copy mode if multiple source buffers */
+	if (count_ready_or_higher_sources(mod_source_info) > 1) {
+		comp_info(dev, "module_adapter_bind(), multi_sink_or_source = true");
+		mod->multi_source_or_sink = true;
```

Comment:
this doesn't seem right. You shouldn't need to count those sources in certain states. You can do this perfectly deterministically on paths where that state is set or cleared. If `.bind()` adds a source in one of those states - then you know that the number has changed. If it adds one in a lower state - it doesn't change here, so no need to re-count. Same with `.unbind()` - do we allow unbinding ready sources?

Reply:
I likely have a bug in unbind since there's fails in repeated suspend-resume test. Need to rethink this, thanks for commenting!

Reply:
@singalsu please take this patch https://github.com/thesofproject/sof/pull/7805/commits/1e8b25299f2c9760c11abe4ffe6256c9ea36d99b and I think your IPC timeouts with unbind will go away

Reply:
Thanks, that solves it! I totally missed that scenario.